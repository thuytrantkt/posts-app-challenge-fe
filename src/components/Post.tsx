import React from "react";
import { PostType } from "../types";

const Post = ({
  post,
  handleSelectedPostClick,
}: {
  post: PostType;
  handleSelectedPostClick: () => void;
}) => {
  return (
    <div
      style={{
        border: "1px solid #ccc",
        borderRadius: "5px",
        padding: "0px 20px",
        cursor: "pointer",
        userSelect: "none",
      }}
      onClick={handleSelectedPostClick}
    >
      <h2
        style={{
          fontSize: "1.5em",
          fontWeight: "700",
          marginBottom: "10px",
        }}
      >
        {post.title}
      </h2>
      <p style={{ marginBottom: "10px" }}>{post.content}</p>
      <p>
        <strong>Author:</strong> {post.author}
      </p>
      <p>
        <strong>Posted on:</strong>{" "}
        {new Date(post.createdAt).toLocaleDateString()}
      </p>
    </div>
  );
};

export default Post;
