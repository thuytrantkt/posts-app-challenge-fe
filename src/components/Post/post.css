.post-container {
  border: 1px solid #ccc;
  border-radius: 5px;
  padding: 0px 20px;
  cursor: pointer;
  user-select: none;
  -webkit-user-select: none;
}

.post-heading {
  font-size: 1.5em;
  font-weight: 700;
  margin-bottom: 10px;
}
