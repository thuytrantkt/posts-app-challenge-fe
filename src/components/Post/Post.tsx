import React from "react";
import { PostProps } from "../../types";
import { dateFormatted } from "../../utils/date-helper";
import "./post.css";

const Post = ({ post, handleSelectedPostClick }: PostProps) => {
  const { author, content, createdAt, title } = post;
  return (
    <div className="post-container" onClick={handleSelectedPostClick}>
      <h2 className="post-heading">{title}</h2>
      <p className="margin-bottom">{content}</p>
      <p>
        <strong>Author:</strong> {author}
      </p>
      <p>
        <strong>Posted on:</strong> {dateFormatted(createdAt)}
      </p>
    </div>
  );
};

export default Post;
