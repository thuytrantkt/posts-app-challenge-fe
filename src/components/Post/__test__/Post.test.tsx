import { render, fireEvent } from "@testing-library/react";
import Post from "../Post";

const postMockData = {
  author: "John Smith",
  commentCount: 4,
  content: "This is my very first post",
  createdAt: "Thu, 19 Aug 2004 16:21:06 GMT",
  id: 1,
  title: "First Post",
};

describe("Post component", () => {
  it("renders Post component properly", () => {
    const view = render(
      <Post post={postMockData} handleSelectedPostClick={() => {}} />
    );
    expect(view).toMatchSnapshot();
  });

  it("should call handleSelectedPostClick when clicked", () => {
    const handleSelectedPostClick = jest.fn();
    const handleSelectedPostKeyDown = jest.fn();

    const { container } = render(
      <Post
        post={postMockData}
        handleSelectedPostClick={handleSelectedPostClick}
        handleSelectedPostKeyDown={handleSelectedPostKeyDown}
      />
    );

    const postDiv = container.firstChild;

    if (postDiv) {
      fireEvent.click(postDiv);
    }

    expect(handleSelectedPostClick).toHaveBeenCalledTimes(1);
  });

  it("should call handleSelectedPostKeyDown when a key is pressed as an 'Enter' key", () => {
    const handleSelectedPostClick = jest.fn();
    const handleSelectedPostKeyDown = jest.fn();

    const { container } = render(
      <Post
        post={postMockData}
        handleSelectedPostClick={handleSelectedPostClick}
        handleSelectedPostKeyDown={handleSelectedPostKeyDown}
      />
    );

    const postDiv = container.firstChild;

    if (postDiv) {
      fireEvent.keyDown(postDiv, { key: "Enter" });
    }

    expect(handleSelectedPostKeyDown).toHaveBeenCalledTimes(1);
  });

  it("should call handleSelectedPostKeyDown when a key is pressed as a space '' key", () => {
    const handleSelectedPostClick = jest.fn();
    const handleSelectedPostKeyDown = jest.fn();

    const { container } = render(
      <Post
        post={postMockData}
        handleSelectedPostClick={handleSelectedPostClick}
        handleSelectedPostKeyDown={handleSelectedPostKeyDown}
      />
    );

    const postDiv = container.firstChild;

    if (postDiv) {
      fireEvent.keyDown(postDiv, { key: " " });
    }

    expect(handleSelectedPostKeyDown).toHaveBeenCalledTimes(1);
  });
});
