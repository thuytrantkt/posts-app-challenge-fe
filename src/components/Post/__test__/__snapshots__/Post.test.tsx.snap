// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post renders Post component properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cursor"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1v5z4dq-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiCardContent-root css-wwt4d9-MuiCardContent-root"
            >
              <h2
                class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1sgfi5h-MuiTypography-root"
              >
                First Post
              </h2>
              <p
                class="MuiTypography-root MuiTypography-body1 css-uriene-MuiTypography-root"
              >
                This is my very first post
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-uriene-MuiTypography-root"
              >
                <strong>
                  Author:
                </strong>
                 
                John Smith
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-uriene-MuiTypography-root"
              >
                <strong>
                  Posted on:
                </strong>
                 
                8/19/2004
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cursor"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1v5z4dq-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiCardContent-root css-wwt4d9-MuiCardContent-root"
          >
            <h2
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1sgfi5h-MuiTypography-root"
            >
              First Post
            </h2>
            <p
              class="MuiTypography-root MuiTypography-body1 css-uriene-MuiTypography-root"
            >
              This is my very first post
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-uriene-MuiTypography-root"
            >
              <strong>
                Author:
              </strong>
               
              John Smith
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-uriene-MuiTypography-root"
            >
              <strong>
                Posted on:
              </strong>
               
              8/19/2004
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
